transport/kubernetes/src/main/kotlin/KubernetesMessageSender.kt

```diff
@@ -141,6 +147,14 @@ internal class KubernetesMessageSender<T : Any>(
         api.createNamespacedJob(msgConfig.namespace, jobBody, null, null, null, null)
     }
 
+    /**
+     * Generate a valid trace ID from the given [message]. Check the ID contained in the message. If it is empty, a
+     * unique ID is generated. This is needed because the job name is derived from the trace ID, and Kubernetes
+     * as some restrictions for such names.
+     */
+    private fun validTraceId(message: Message<T>): String =
+        message.header.traceId.takeUnless { it.isBlank() } ?: "undefined-${nameCounter.incrementAndGet()}"
```

Comment:
Is the trace id supposed to be unique ?
Could you not get collisions if core (orcherstrator?) pod is evicted and recreated, of if the cluster restart ?

Why not use some UUIDs instead of the counter ?

Reply:
Switched to UUIDs.